controllers/integrationpipeline/integrationpipeline_adapter.go

```diff
@@ -129,6 +129,7 @@ func (a *Adapter) EnsureSnapshotPassedAllTests() (controller.OperationResult, er
 				"The global component list has changed in the meantime, superseding with a composite snapshot")
 			a.logger.LogAuditEvent("Snapshot integration status condition marked as invalid, the global component list has changed in the meantime",
 				existingSnapshot, h.LogActionUpdate)
+			return controller.ContinueProcessing()
```

Comment:
The issue with this change is that the `SetSnapshotIntegrationStatusAsInvalid()` function doesn't update the resource on the cluster, so the Snapshot CR will never be updated/patched like it happens when calling `MarkSnapshotAs*` functions.
Probably the simplest solution would be to add an additional patch call here.

Reply:
Thanks! Added the patching code as you suggested. Can you please help to review it again?

Reply:
Thanks for the change, can you also move the `a.logger.LogAuditEvent` call to happen after the resource is updated(patched)?

Reply:
OK.

Reply:
Thanks!